Core/SymbolValidation/SymbolValidator.cs

```diff
@@ -23,18 +23,25 @@ public class SymbolValidator
         private readonly string _packagePath;
         private readonly IFolder _rootFolder;
         private readonly HttpClient _httpClient;
+        private readonly ITemporaryFileProvider? _tempProvider;
 
         public SymbolValidator(IPackage package, string packagePath, IFolder? rootFolder = null)
             : this(package, packagePath, rootFolder, httpClient: null)
         {
         }
 
         public SymbolValidator(IPackage package, string packagePath, IFolder? rootFolder, HttpClient? httpClient)
+            : this(package, packagePath, rootFolder, httpClient, tempProvider: null)
+        {
+        }
+
+        public SymbolValidator(IPackage package, string packagePath, IFolder? rootFolder, HttpClient? httpClient, ITemporaryFileProvider? tempProvider)
```

Comment:
Would it be better to have a default implementation of `ITemporaryFileProvider` registered in DI instead of null checks and if statements later?

Reply:
I didn't want to make assumptions on what the temp file provider would do and instead leave the fully up to the user. Maybe if we get feedback from folks using the Core package on common usage patterns?